src/embeddedTraceViewer.ts

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { TestConfig } from './playwrightTestTypes';
+import { getNonce } from './utils';
+import * as vscodeTypes from './vscodeTypes';
+import { DisposableBase } from './disposableBase';
+import { PlaywrightTestServer } from './playwrightTestServer';
+import { TraceViewer } from './traceViewer';
+import { SettingsModel } from './settingsModel';
+import { PlaywrightTestCLI } from './playwrightTestCLI';
+
+export class EmbeddedTraceViewer implements TraceViewer {
+  readonly vscode: vscodeTypes.VSCode;
+  readonly extensionUri: vscodeTypes.Uri;
+  private _settingsModel: SettingsModel;
+  private _currentFile?: string;
+  private _traceViewerPanel: EmbeddedTraceViewerPanel | undefined;
+  private _traceLoadRequestedTimeout?: NodeJS.Timeout;
+  private _config: TestConfig;
+  private _testServer?: PlaywrightTestServer;
```

Comment:
I don't think embedded trace viewer makes sense w/o the test server, make it required?

Reply:
I was assuming that the **Embedded** checkbox would be available even in legacy versions that had `PlaywrightTestCLI`, and that way the `EmbeddedTraceViewer` instance would show the warning message.

But if the warning message is not needed, I can make it required and only instanciate `EmbeddedTraceViewer` when it makes sense. In that case, I can also move the `checkSupport` logic into `testModel` and remove `settingsModel` dependency from trace viewer.

Reply:
Yes, let's only support Embedded for modern PlaywrightTestServer-based configurations only.

Reply:
Addressed in 31a15f3f926c5f17a8e16a4476d52c62f11778a9